packages/did-document/src/did-document.ts

```diff
@@ -101,10 +103,10 @@ export default class DIDDocument {
             return !entry.id.match(contextDid)
         })
         this.doc.assertionMethod = this.doc.assertionMethod!.filter((entry: string | VerificationMethod) => {
-            return entry !== `${this.doc.id}?context=${contextHash}#sign`
+            return entry !== `${this.doc.id}?context=${contextHash}`
```

Comment:
![image](https://user-images.githubusercontent.com/15656252/192335236-d2004a93-0189-4c12-afb8-ab83b13e40a8.png)
Please take a look for Key purpose of public keys.
As there are purpose of keys while adding by setAttribute() function, we don't need feel to set key type after context.
As you can see, there are 3 key purposes of veriKey, sigAuth, & enc.
Here I matched "#sign" to sigAuth & "#asym" to veriKey.
Do we need to add more key types?
And I am not sure whether adding more types is compatible with DID spec.

Reply:
Link is https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md

Reply:
I have resolved this issue with this commit: https://github.com/verida/verida-js/pull/211/commits/b069bd590456a95c9134cd04f23994583fd545ed

Reply:
Actually, I haven't resolved this.

How can `did-document` specify the key purpose so that it is saved? I think that's the problem. This code needs to set the key purpose so that `vda-did-resolver` can save it to the blockchain.

Reply:
Please take a look line#93 ~ line#100 of did-client/src/helpers.ts.
You can know that how it deduce the keyPurpose from created DIDDocument.
For each item in the document.verificationMethod, if same id is contained in document.authentication the keyPurpose is 'sigAuth'. If same id is contained in document.keyAgreemtn, keyPurpose is 'enc'.

![image](https://user-images.githubusercontent.com/15656252/193190373-f959c8f5-21d2-44bd-8dc5-4e5a8cd5d9cb.png)

Reply:
How does this map back to the DID document when the smart contract is read?

Can we make this more explicit by using query parameters in the publicKey that specify “typ”? That logic will not always hold true in the future

Reply:
Please check line#266 ~ line#270 in the vda-did-resolver/src/resolver.ts. Same logic. I created above code from the vda-did-resolver. They're matched.

![image](https://user-images.githubusercontent.com/15656252/193191839-485fca17-5805-4a70-ae33-d8004cca14de.png)

Reply:
Of course, we can add type in the attribute value. To do so, need to check out vda-did & vda-did-resolver again.